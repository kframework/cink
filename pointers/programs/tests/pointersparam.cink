void swap(int *p, int *q) {
  int a;
  a = *p;
  *p = *q;
  *q = a;
}

int main() {
  int a = 3, b= 4;
  int *x = &a, *y = &b;
  swap(x, y);
  cout << a << " " << b << endl;
  return 0;
}

