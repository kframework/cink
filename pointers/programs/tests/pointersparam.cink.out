4 3
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            0
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(2)
            swap |-> #symScalLoc(1)
        </env>
    </control>
    <store>
        #symScalLoc(1) |-> lambda ( int (* p), int (* q) @ int a ; (a = (*
           p)) ; ((* p) = (* q)) ; ((* q) = a) ; )
        #symScalLoc(2) |-> lambda (  @ int a ; (a = 3) ; int b ; (b = 4) ; x 
          of pointer to int ; (x = (& a)) ; y of pointer to int ; (y = (& b)) 
          ; (swap ( (x , y) )) ; ((((cout << a) << " ") << b) << (endl)) ; 
          return 0 ; )
        #symScalLoc(3) |-> 4
        #symScalLoc(4) |-> 3
        #symScalLoc(5) |-> #symScalLoc(3)
        #symScalLoc(6) |-> #symScalLoc(4)
        #symScalLoc(7) |-> #symScalLoc(3)
        #symScalLoc(8) |-> #symScalLoc(4)
        #symScalLoc(9) |-> 3
    </store>
    <genv>
        main |-> #symScalLoc(2)
        swap |-> #symScalLoc(1)
    </genv>
</T>
